float64 p0
float64 pf
float64 v0
float64 vf
float64 t0
float64 tf
